Docker build fails for me

System: NixOS unstable.

```
$ make docker-build
docker build -t cuekeeper .
[+] Building 121.0s (10/11)                                                         docker:default
 => [internal] load build definition from Dockerfile                                          0.0s
 => => transferring dockerfile: 459B                                                          0.0s
 => [internal] load metadata for docker.io/ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc  2.5s
 => [internal] load .dockerignore                                                             0.0s
 => => transferring context: 98B                                                              0.0s
 => [1/7] FROM docker.io/ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc195b29f1006d0f97a  76.5s
 => => resolve docker.io/ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc195b29f1006d0f97ae  0.0s
 => => sha256:2f9a9fc195b29f1006d0f97ae4c6300fb175800703b12a15b51583a1f7a547 1.36kB / 1.36kB  0.0s
 => => sha256:0e699b4ca5858785207e0ad168b3aff3fbd29ae35627c229000ac0ce8f23562b 955B / 955B    0.0s
 => => sha256:e1e562a55a4d1c6a1a35929771675af0d96761f192a8021e03901faafed776 9.85kB / 9.85kB  0.0s
 => => sha256:8f0fdd3eaac017684c880d836abdaf02f9db7ac8ecca970356482e3d8e31 50.38MB / 50.38MB  7.4s
 => => sha256:f8589b1e23e8c0abd00edb7edd80c395c0acf87eac2c06c9236ca94c2 244.38MB / 244.38MB  37.2s
 => => sha256:96ef65255281003578a4de6eb3ed4f8ec29f615ca492873c11564e750 119.60MB / 119.60MB  27.8s
 => => extracting sha256:8f0fdd3eaac017684c880d836abdaf02f9db7ac8ecca970356482e3d8e315650     3.9s
 => => extracting sha256:f8589b1e23e8c0abd00edb7edd80c395c0acf87eac2c06c9236ca94c216d49ae    20.0s
 => => extracting sha256:96ef65255281003578a4de6eb3ed4f8ec29f615ca492873c11564e750ea640c6    16.8s
 => [internal] load build context                                                             0.0s
 => => transferring context: 1.43kB                                                           0.0s
 => [2/7] RUN mkdir /home/opam/cuekeeper                                                      0.6s
 => [3/7] COPY --chown=opam cuekeeper.opam /home/opam/cuekeeper/                              0.1s
 => [4/7] WORKDIR /home/opam/cuekeeper                                                        0.1s
 => [5/7] RUN opam pin add -yn cuekeeper.dev .                                                6.0s
 => ERROR [6/7] RUN opam depext -t cuekeeper                                                 35.1s
------                                                                                             
 > [6/7] RUN opam depext -t cuekeeper:                                                             
0.403 # Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=debian, os-family=debian                                                                                                 
32.21 libgmp-dev                                                                                   
32.21 m4                                                                                           
32.21 perl
32.21 # The following system packages are needed:
33.79 The following command needs to be run through "sudo":
33.79     apt-get install -qq -yy libgmp-dev m4 perl
35.07 E: Failed to fetch http://deb.debian.org/debian/pool/main/g/gmp/libgmpxx4ldbl_6.1.2+dfsg-4_amd64.deb  404  Not Found [IP: 151.101.2.132 80]
35.07 E: Failed to fetch http://deb.debian.org/debian/pool/main/g/gmp/libgmp-dev_6.1.2+dfsg-4_amd64.deb  404  Not Found [IP: 151.101.2.132 80]
35.07 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
35.08 OS package installation failed
------
Dockerfile:7
--------------------
   5 |     WORKDIR /home/opam/cuekeeper
   6 |     RUN opam pin add -yn cuekeeper.dev .
   7 | >>> RUN opam depext -t cuekeeper
   8 |     RUN opam install -t --deps-only cuekeeper
   9 |     ENTRYPOINT ["opam", "config", "exec", "--"]
--------------------
ERROR: failed to solve: process "/bin/sh -c opam depext -t cuekeeper" did not complete successfully: exit code: 1
make: *** [Makefile:37: docker-build] Error 1
```

I tried changing Dockerfile to do apt update (which I think is quite dirty):

```
FROM ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc195b29f1006d0f97ae4c6300fb175800703b12a15b51583a1f7a547b7
#FROM ocurrent/opam:debian-10-ocaml-4.07
RUN mkdir /home/opam/cuekeeper
COPY --chown=opam cuekeeper.opam /home/opam/cuekeeper/
WORKDIR /home/opam/cuekeeper

# (My change)
RUN sudo apt update --allow-releaseinfo-change -y

RUN opam pin add -yn cuekeeper.dev .
RUN opam depext -t cuekeeper
RUN opam install -t --deps-only cuekeeper
ENTRYPOINT ["opam", "config", "exec", "--"]
```

Then I get the following error:

```
docker build -t cuekeeper .
[+] Building 34.1s (12/12) FINISHED                                                                      docker:default
 => [internal] load build definition from Dockerfile                                                               0.0s
 => => transferring dockerfile: 521B                                                                               0.0s
 => [internal] load metadata for docker.io/ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc195b29f1006d0f97ae4c6  0.0s
 => [internal] load .dockerignore                                                                                  0.0s
 => => transferring context: 98B                                                                                   0.0s
 => [1/8] FROM docker.io/ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc195b29f1006d0f97ae4c6300fb175800703b12a  0.0s
 => [internal] load build context                                                                                  0.0s
 => => transferring context: 36B                                                                                   0.0s
 => CACHED [2/8] RUN mkdir /home/opam/cuekeeper                                                                    0.0s
 => CACHED [3/8] COPY --chown=opam cuekeeper.opam /home/opam/cuekeeper/                                            0.0s
 => CACHED [4/8] WORKDIR /home/opam/cuekeeper                                                                      0.0s
 => CACHED [5/8] RUN sudo apt update --allow-releaseinfo-change -y                                                 0.0s
 => CACHED [6/8] RUN opam pin add -yn cuekeeper.dev .                                                              0.0s
 => CACHED [7/8] RUN opam depext -t cuekeeper                                                                      0.0s
 => ERROR [8/8] RUN opam install -t --deps-only cuekeeper                                                         34.0s
------                                                                                                                  
 > [8/8] RUN opam install -t --deps-only cuekeeper:                                                                     
0.373 [NOTE] It seems you have not updated your repositories for a while. Consider updating them with:                  
0.373        opam update                                                                                                
0.373                                                                                                                   
0.373                                                                                                                   
0.373 <><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
0.397 [cuekeeper.dev] no changes from file:///home/opam/cuekeeper
0.398 
2.579 The following actions will be performed:
2.580   - install seq                     base
2.580   - install conf-m4                 1
2.580   - install ocamlbuild              0.14.0
2.580   - install cmdliner                1.0.4
2.580   - install dune                    2.1.1
2.580   - install jbuilder                1.0+beta20.2
2.580   - install ocamlfind               1.8.1
2.580   - install uchar                   0.0.2
2.580   - install stdlib-shims            0.1.0
2.580   - install sexplib0                v0.13.0
2.580   - install result                  1.4
2.580   - install re                      1.9.0
2.580   - install ppx_derivers            1.2.1
2.580   - install ocaml-compiler-libs     v0.12.1
2.580   - install mmap                    1.1.0
2.580   - install easy-format             1.3.2
2.580   - install dune-private-libs       2.1.1
2.580   - install cppo                    1.6.6
2.580   - install bigarray-compat         1.0.0
2.580   - install topkg                   1.0.1
2.580   - install ppx_tools               5.1+4.06.0
2.580   - install ocamlgraph              1.8.8
2.580   - install num                     1.3
2.580   - install base-bytes              base
2.580   - install ocaml-migrate-parsetree 1.5.0
2.580   - install biniou                  1.2.1
2.580   - install dune-configurator       2.1.1
2.580   - install cstruct                 5.1.1
2.581   - install uutf                    1.0.2
2.581   - install uuidm                   0.9.7
2.581   - install react                   1.2.1
2.581   - install fmt                     0.8.8
2.581   - install stringext               1.6.0
2.581   - install ounit2                  2.2.1
2.581   - install omd                     1.3.1
2.581   - install ocplib-endian           1.0
2.581   - install decompress              0.7
2.581   - install base64                  3.2.0
2.581   - install astring                 0.8.3
2.581   - install ppxfind                 1.3
2.581   - install ppx_tools_versioned     5.2.3
2.581   - install yojson                  1.7.0
2.581   - install base                    v0.13.0
2.581   - install mstruct                 1.4.0
2.581   - install hex                     1.4.0
2.581   - install tyxml                   4.3.0
2.581   - install jsonm                   1.0.1
2.581   - install reactiveData            0.2.2
2.581   - install uri                     3.1.0
2.581   - install ounit                   2.2.1
2.581   - install lwt                     5.1.1
2.581   - install ppx_deriving            4.4
2.581   - install js_of_ocaml-compiler    3.5.2
2.581   - install stdio                   v0.13.0
2.581   - install parsexp                 v0.13.0
2.581   - install fieldslib               v0.13.0
2.581   - install js_of_ocaml             3.5.2
2.581   - install ppxlib                  0.11.0
2.581   - install sexplib                 v0.13.0
2.581   - install ptime                   0.8.5
2.581   - install logs                    0.7.0
2.581   - install js_of_ocaml-ppx         3.5.2
2.581   - install ppx_sexp_conv           v0.13.0
2.581   - install ppx_fields_conv         v0.13.0
2.581   - install ppx_cstruct             5.1.1
2.581   - install crunch                  3.2.0
2.581   - install irmin                   1.4.0
2.581   - install git                     1.11.5
2.581   - install js_of_ocaml-tyxml       3.5.2
2.581   - install js_of_ocaml-lwt         3.5.2
2.581   - install uri-sexp                3.1.0
2.581   - install tar                     1.1.0
2.581   - install irmin-mem               1.3.0
2.582   - install irmin-git               1.4.0*
2.582   - install cohttp                  2.5.0
2.582   - install irmin-indexeddb         1.3*
2.582   - install cohttp-lwt              2.5.0
2.582   - install cohttp-lwt-jsoo         2.5.0
2.582 ===== 78 to install =====
2.582 
2.582 <><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
3.229 [astring.0.8.3] downloaded from http://erratique.ch/software/astring/releases/astring-0.8.3.tbz
3.331 [base.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/base-v0.13.0.tar.gz
3.804 [base64.3.2.0] downloaded from https://github.com/mirage/ocaml-base64/releases/download/v3.2.0/base64-v3.2.0.tbz
4.429 [bigarray-compat.1.0.0] downloaded from https://github.com/mirage/bigarray-compat/archive/v1.0.0.tar.gz
4.517 [cmdliner.1.0.4] downloaded from http://erratique.ch/software/cmdliner/releases/cmdliner-1.0.4.tbz
4.895 [biniou.1.2.1] downloaded from https://github.com/mjambon/biniou/releases/download/1.2.1/biniou-1.2.1.tbz
6.047 [cohttp-lwt.2.5.0] downloaded from https://github.com/mirage/ocaml-cohttp/releases/download/v2.5.0/cohttp-v2.5.0.tbz
6.198 [cohttp.2.5.0] downloaded from https://github.com/mirage/ocaml-cohttp/releases/download/v2.5.0/cohttp-v2.5.0.tbz
6.353 [cohttp-lwt-jsoo.2.5.0] downloaded from https://github.com/mirage/ocaml-cohttp/releases/download/v2.5.0/cohttp-v2.5.0.tbz
7.690 [crunch.3.2.0] downloaded from https://github.com/mirage/ocaml-crunch/releases/download/v3.2.0/crunch-v3.2.0.tbz
7.841 [cppo.1.6.6] downloaded from https://github.com/ocaml-community/cppo/releases/download/v1.6.6/cppo-v1.6.6.tbz
7.997 [cstruct.5.1.1] downloaded from https://github.com/mirage/ocaml-cstruct/releases/download/v5.1.1/cstruct-v5.1.1.tbz
9.214 [decompress.0.7] downloaded from https://github.com/mirage/decompress/releases/download/v0.7/decompress-0.7.tbz
9.643 [dune.2.1.1] downloaded from https://github.com/ocaml/dune/releases/download/2.1.1/dune-2.1.1.tbz
9.797 [dune-configurator.2.1.1] downloaded from https://github.com/ocaml/dune/releases/download/2.1.1/dune-2.1.1.tbz
10.74 [dune-private-libs.2.1.1] downloaded from https://github.com/ocaml/dune/releases/download/2.1.1/dune-2.1.1.tbz
12.15 [easy-format.1.3.2] downloaded from https://github.com/mjambon/easy-format/releases/download/1.3.2/easy-format-1.3.2.tbz
12.24 [fieldslib.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/fieldslib-v0.13.0.tar.gz
12.43 [fmt.0.8.8] downloaded from https://erratique.ch/software/fmt/releases/fmt-0.8.8.tbz
13.51 [hex.1.4.0] downloaded from https://github.com/mirage/ocaml-hex/releases/download/v1.4.0/hex-v1.4.0.tbz
13.59 [git.1.11.5] downloaded from https://github.com/mirage/ocaml-git/releases/download/1.11.5/git-1.11.5.tbz
13.80 [irmin.1.4.0] downloaded from https://github.com/mirage/irmin/releases/download/1.4.0/irmin-1.4.0.tbz
15.28 [irmin-mem.1.3.0] downloaded from https://github.com/mirage/irmin/releases/download/1.3.0/irmin-1.3.0.tbz
15.43 [irmin-git.1.4.0] synchronised from git+https://github.com/talex5/irmin.git#dedb0dd44b70806b1ab68c666c140c27caebe1c2
15.91 [irmin-indexeddb.1.3] synchronised from git+https://github.com/talex5/irmin-indexeddb.git#c13c3e489af66868ddd86be37c958e56375db43b
16.62 [jbuilder.1.0+beta20.2] downloaded from https://github.com/ocaml/dune/releases/download/1.0%2Bbeta20.2/jbuilder-1.0+beta20.2.tbz
16.84 [js_of_ocaml.3.5.2] downloaded from https://github.com/ocsigen/js_of_ocaml/releases/download/3.5.2/js_of_ocaml-3.5.2.tbz
17.07 [js_of_ocaml-lwt.3.5.2] found in cache
17.28 [js_of_ocaml-ppx.3.5.2] found in cache
17.40 [js_of_ocaml-compiler.3.5.2] downloaded from https://github.com/ocsigen/js_of_ocaml/releases/download/3.5.2/js_of_ocaml-3.5.2.tbz
17.63 [js_of_ocaml-tyxml.3.5.2] found in cache
18.31 [jsonm.1.0.1] downloaded from http://erratique.ch/software/jsonm/releases/jsonm-1.0.1.tbz
18.40 [logs.0.7.0] downloaded from https://erratique.ch/software/logs/releases/logs-0.7.0.tbz
19.66 [mstruct.1.4.0] downloaded from https://github.com/mirage/ocaml-mstruct/releases/download/v1.4.0/mstruct-1.4.0.tbz
19.74 [mmap.1.1.0] downloaded from https://github.com/mirage/mmap/releases/download/v1.1.0/mmap-v1.1.0.tbz
19.94 [lwt.5.1.1] downloaded from https://github.com/ocsigen/lwt/archive/5.1.1.tar.gz
20.80 [ocaml-compiler-libs.v0.12.1] downloaded from https://github.com/janestreet/ocaml-compiler-libs/archive/v0.12.1.tar.gz
20.89 [num.1.3] downloaded from https://github.com/ocaml/num/archive/v1.3.tar.gz
21.38 [ocaml-migrate-parsetree.1.5.0] downloaded from https://github.com/ocaml-ppx/ocaml-migrate-parsetree/releases/download/v1.5.0/ocaml-migrate-parsetree-v1.5.0.tbz
21.64 [ocamlfind.1.8.1] downloaded from http://download.camlcity.org/download/findlib-1.8.1.tar.gz
22.19 [ocamlbuild.0.14.0] downloaded from https://github.com/ocaml/ocamlbuild/archive/0.14.0.tar.gz
22.84 [ocplib-endian.1.0] downloaded from https://github.com/OCamlPro/ocplib-endian/archive/1.0.tar.gz
23.38 [omd.1.3.1] downloaded from https://github.com/Chris00/omd/releases/download/1.3.1/omd-1.3.1.tar.gz
23.60 [ounit.2.2.1] downloaded from https://github.com/gildor478/ounit/releases/download/v2.2.1/ounit-v2.2.1.tbz
23.74 [ppx_cstruct.5.1.1] found in cache
24.03 [ounit2.2.2.1] downloaded from https://github.com/gildor478/ounit/releases/download/v2.2.1/ounit-v2.2.1.tbz
24.11 [parsexp.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/parsexp-v0.13.0.tar.gz
24.68 [ppx_derivers.1.2.1] downloaded from https://github.com/ocaml-ppx/ppx_derivers/archive/1.2.1.tar.gz
24.76 [ppx_fields_conv.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/ppx_fields_conv-v0.13.0.tar.gz
25.36 [ppx_sexp_conv.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/ppx_sexp_conv-v0.13.0.tar.gz
25.62 [ppx_deriving.4.4] downloaded from https://github.com/ocaml-ppx/ppx_deriving/archive/v4.4.tar.gz
25.91 [ppx_tools.5.1+4.06.0] downloaded from https://github.com/ocaml-ppx/ppx_tools/archive/5.1+4.06.0.tar.gz
26.48 [ppx_tools_versioned.5.2.3] downloaded from https://github.com/ocaml-ppx/ppx_tools_versioned/archive/5.2.3.tar.gz
26.57 [ptime.0.8.5] downloaded from https://erratique.ch/software/ptime/releases/ptime-0.8.5.tbz
27.14 [ppxlib.0.11.0] downloaded from https://github.com/ocaml-ppx/ppxlib/archive/0.11.0.tar.gz
27.24 [react.1.2.1] downloaded from http://erratique.ch/software/react/releases/react-1.2.1.tbz
27.85 [re.1.9.0] downloaded from https://github.com/ocaml/ocaml-re/releases/download/1.9.0/re-1.9.0.tbz
28.28 [reactiveData.0.2.2] downloaded from https://github.com/ocsigen/reactiveData/archive/0.2.2.tar.gz
28.37 [result.1.4] downloaded from https://github.com/janestreet/result/archive/1.4.tar.gz
28.58 [sexplib.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/sexplib-v0.13.0.tar.gz
28.93 [stdio.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/stdio-v0.13.0.tar.gz
29.02 [sexplib0.v0.13.0] downloaded from https://ocaml.janestreet.com/ocaml-core/v0.13/files/sexplib0-v0.13.0.tar.gz
29.88 [stdlib-shims.0.1.0] downloaded from https://github.com/ocaml/stdlib-shims/releases/download/0.1.0/stdlib-shims-0.1.0.tbz
30.36 [stringext.1.6.0] downloaded from https://github.com/rgrinberg/stringext/releases/download/1.6.0/stringext-1.6.0.tbz
30.52 [tar.1.1.0] downloaded from https://github.com/mirage/ocaml-tar/releases/download/v1.1.0/tar-v1.1.0.tbz
30.61 [topkg.1.0.1] downloaded from http://erratique.ch/software/topkg/releases/topkg-1.0.1.tbz
31.90 [tyxml.4.3.0] downloaded from https://github.com/ocsigen/tyxml/releases/download/4.3.0/tyxml-4.3.0.tbz
31.98 [uchar.0.0.2] downloaded from https://github.com/ocaml/uchar/releases/download/v0.0.2/uchar-0.0.2.tbz
32.13 [uri.3.1.0] downloaded from https://github.com/mirage/ocaml-uri/releases/download/v3.1.0/uri-v3.1.0.tbz
32.26 [uri-sexp.3.1.0] found in cache
32.64 [uuidm.0.9.7] downloaded from https://erratique.ch/software/uuidm/releases/uuidm-0.9.7.tbz
32.89 [uutf.1.0.2] downloaded from http://erratique.ch/software/uutf/releases/uutf-1.0.2.tbz
33.56 [yojson.1.7.0] downloaded from https://github.com/ocaml-community/yojson/releases/download/1.7.0/yojson-1.7.0.tbz
33.67 [ERROR] The sources of the following couldn't be obtained, aborting:
33.67           - ocamlgraph.1.8.8: curl error code 404
33.67           - ppxfind.1.3: curl error code 404
33.67 
------
Dockerfile:12
--------------------
  10 |     RUN opam pin add -yn cuekeeper.dev .
  11 |     RUN opam depext -t cuekeeper
  12 | >>> RUN opam install -t --deps-only cuekeeper
  13 |     ENTRYPOINT ["opam", "config", "exec", "--"]
  14 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c opam install -t --deps-only cuekeeper" did not complete successfully: exit code: 40
make: *** [Makefile:37: docker-build] Error 1
```

I tried adding `opam update` as well (which I think is also quite dirty), but the error stayed the same.

Do you still use Cuekeeper, by the way?

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,9 +1,10 @@
-FROM ocurrent/opam:debian-10-ocaml-4.07@sha256:2f9a9fc195b29f1006d0f97ae4c6300fb175800703b12a15b51583a1f7a547b7
-#FROM ocurrent/opam:debian-10-ocaml-4.07
+FROM ocaml/opam:debian-12-ocaml-4.07@sha256:a887442d9759c6062440ede489808ac5a381efa0b7048582bc45d813de826ea7
+#FROM ocurrent/opam:debian-12-ocaml-4.07
+# Make sure we're using opam-2.2:
+RUN sudo ln -sf /usr/bin/opam-2.2 /usr/bin/opam
 RUN mkdir /home/opam/cuekeeper
 COPY --chown=opam cuekeeper.opam /home/opam/cuekeeper/
 WORKDIR /home/opam/cuekeeper
 RUN opam pin add -yn cuekeeper.dev .
-RUN opam depext -t cuekeeper
 RUN opam install -t --deps-only cuekeeper
-ENTRYPOINT ["opam", "config", "exec", "--"]
+ENTRYPOINT ["opam", "exec", "--"]

--- a/cuekeeper.opam
+++ b/cuekeeper.opam
@@ -11,12 +11,12 @@ depends: [
   "sexplib" {>= "v0.13.0"}
   "uuidm"
   "irmin"
-  "tyxml" {>= "4.3.0"}
+  "tyxml" {>= "4.6.0"}
   "reactiveData" {>= "0.2"}
-  "js_of_ocaml" {>= "3.0.0"}
+  "js_of_ocaml" {>= "4.0.0"}
   "js_of_ocaml-tyxml"
-  "omd"
-  "fmt"
+  "omd" {>= "1.2.4" & < "2.0.0~"}
+  "fmt" {>= "0.8.9" }
   "logs"
   "irmin-mem" {with-test}
   "ounit" {with-test}

--- a/js/ck_animate.ml
+++ b/js/ck_animate.ml
@@ -18,6 +18,10 @@ let async : (unit -> unit Lwt.t) -> unit = Lwt_js_events.async
 let clamp lo hi v =
   min (max v lo) hi
 
+let request_animation_frame f =
+  let callback = Js.wrap_callback (fun (_ : float) -> f ()) in
+  ignore (Dom_html.window##requestAnimationFrame callback : Dom_html.animation_frame_request_id)
+
 (* Fade out, then animate max-height down to zero. Can't do this with CSS because of FF bug #830056.
  * [when_complete] is called on success (but not if cancelled). *)
 let fade_out ?when_complete elem =
@@ -37,7 +41,7 @@ let fade_out ?when_complete elem =
       elem##.style##.opacity := Js.string o |> Js.Optdef.return;
       elem##.style##.maxHeight := (Js.string (string_of_int h ^ "px"));
       if h > 0 then
-        Dom_html._requestAnimationFrame (Js.wrap_callback aux)
+        request_animation_frame aux
       else 
         match when_complete with
         | None -> ()
@@ -71,7 +75,7 @@ let fade_in_move ~full_height elem =
         elem##.style##.opacity := Js.string o |> Js.Optdef.return;
         elem##.style##.maxHeight := (Js.string (string_of_int h ^ "px")) in
       if o < 1.0 then
-        Dom_html._requestAnimationFrame (Js.wrap_callback aux)
+        request_animation_frame aux
     ) in
   async (fun () -> Lwt_js.sleep fade_time >|= fun () ->
     (* By this time, we've had a chance to fill in the height of the removed item. *)
@@ -94,7 +98,7 @@ let animate_scroll_to (target_x, target_y) =
       let dy = float_of_int (target_y - start_y) *. f in
       root##.scrollLeft := start_x + truncate dx;
       root##.scrollTop := start_y + truncate dy;
-      if f < 1.0 then Dom_html._requestAnimationFrame (Js.wrap_callback aux) in
+      if f < 1.0 then request_animation_frame aux in
     aux ()
   )
 

--- a/js/ck_template.ml
+++ b/js/ck_template.ml
@@ -1200,7 +1200,7 @@ module Make (M : Ck_model_s.MODEL with type gui_data = Gui_tree_data.t) = struct
         )
       );
       true in
-    let input_box = input ~a:[a_oninput oninput; a_name "v"; a_placeholder "Add or search"; a_size 20; a_autocomplete false] () in
+    let input_box = input ~a:[a_oninput oninput; a_name "v"; a_placeholder "Add or search"; a_size 20; a_autocomplete `Off] () in
     let add adder _ev =
       match React.S.value value with
       | "" -> ()

--- a/js/client.ml
+++ b/js/client.ml
@@ -31,7 +31,7 @@ module M = Ck_model.Make(Clock)(Git)(Ck_template.Gui_tree_data)(Ck_authn_RPC)
 module T = Ck_template.Make(M)
 
 let server =
-  let use_server = Js.Unsafe.variable "ck_use_server" |> Js.to_bool in
+  let use_server = Js.Unsafe.pure_js_expr "ck_use_server" |> Js.to_bool in
   if use_server then Some (
     Js.to_string Dom_html.window##.location##.protocol ^ Js.to_string Dom_html.window##.location##.host
     |> Uri.of_string

